include/picongpu/algorithms/FieldToParticleInterpolation.hpp

```diff
@@ -29,6 +29,26 @@
 namespace picongpu
 {
 
+template<typename T_Type>
+struct GetComponent
+{
+    using Type = T_Type;
+    using result_type = Type;
+    uint32_t component;
+
+    HDINLINE GetComponent(const GetComponent&) = default;
+    HDINLINE GetComponent(const uint32_t component) : component(component) {}
+    HDINLINE GetComponent& operator=(const GetComponent& other) {
```

Comment:
Is this operator still needed since we removed the const from `component`. see  [this](https://github.com/ComputationalRadiationPhysics/picongpu/pull/2150/files#diff-54b2a2217632dae0dd703dfbdf37b113R59) implementation